ui.tests/test-module/specs/panelcontainer/panelcontainer.runtime.spec.js

```diff
@@ -0,0 +1,161 @@
+/*******************************************************************************
+ * Copyright 2022 Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+describe("Form with Panel Container", () => {
+
+    const pagePath = "content/forms/af/core-components-it/samples/panelcontainer/basic.html";
+    const childBemBlock = 'cmp-adaptiveform-datepicker';
+
+    let formContainer = null
+
+    beforeEach(() => {
+        cy.previewForm(pagePath).then(p => {
+            formContainer = p;
+        })
+    });
+
+    const checkView = (id, view, count) => {
+        expect(view.children.length, "panel has children equal to count").to.equal(count);
+    }
+
+    const checkViewUpdate = (id, view, count) => {
+        //adding delay for view update via mutation observer after html is appended
```

Comment:
@rismehta This test pass in browser mode and fails in headless. cy.wait isn't effective. Any pointers how to handle such mutation observer based tests in headless.